### DISCUSSION TITLE
Add missing actions for Particle Emitter properties

### DISCUSSION THREAD

**Discussion by tristanbob** [4ian/GDevelop#3654] [Enhancement ideas (and their technical discussions)]
## Description

Particle emitters are missing actions for some properties.

### New Actions:

- [ ] Particle type (Circle/Line/Image)  
- [ ] Circle radius (pixles)
- [ ] Line length (pixels)
- [ ] Line thickness (pixels)
- [ ] Select an image (image)
- [x] Additive rendering (bool)
- [ ] Delete when out of particles (bool)
- [x] Maximum number of particles displayed (number)
- [x] Particle minimum rotation speed (degrees/second)
- [x] Particle maximum rotation speed (degrees/second)

### Mismatched:

Gravity actions and properties do not appear to be referring to the same thing.  (My preference would be to change both the properties and actions to use the "angle and speed" method which makes it easy for the user to pull/push particles in a direction.)

  - Actions
    - Gravity 
    - Gravity Angle
  - Properties
    - Gravity on particles on X axis
    - Gravity on particles on Y axis

![ParticleEmitterCommon](https://user-images.githubusercontent.com/8879811/104856401-bef2f480-58cf-11eb-942a-26af915bccb7.PNG)

![ParticleEmitterAdvanced](https://user-images.githubusercontent.com/8879811/104856404-c1554e80-58cf-11eb-8d48-d12931b61022.PNG)

![Capture](https://user-images.githubusercontent.com/8879811/104856408-c4503f00-58cf-11eb-8b9c-54408cc045e7.PNG)

**Comment by tristanbob** [comment]
FYI - I need these actions so that I can build a "Particle Emitter Builder" that lets you see how the parameters affect the look, in real-time.   

Once we have a Particle Emitter Builder, we can create a contest for people to build new emitters!

**Comment by Oxey405** [comment]
Okay it's seems a little hard to implement for a noob like me but still can try ^^
I think this is it :
GDevelop/Extensions/ParticleSystem/Extension.cpp

**Comment by Oxey405** [comment]
Oh my gosh what a hell to get inside this part of the code !! I can't understand !

**Comment by 4ian** [comment]
@Oxey405 Please use the Discord/forum to discuss or share your comments, GitHub is only for constructed collaboration with clear question/answers. Thanks!

**Comment by Oxey405** [comment]
Ok... I give up ! I can't understand !

**Reply by AlexandreSi** [reply]
If you ever want to try again, I can help with some indications

**Comment by 85ayush** [comment]
I would love to work on this issue, I am new to open source contribution, can you please guide me?

**Comment by AlexandreSi** [comment]
I ticked the following actions from your list (done with #4419):
- Additive rendering (bool)
- Maximum number of particles displayed (number)
- Particle minimum rotation speed (degrees/second)
- Particle maximum rotation speed (degrees/second)

The actions regarding the particle (type and shape) are not straightforward as we have to decide how it would be done: Is it a single action that can set the particle or should we have 3 different actions?
But they're doable.

The action to set the option to delete when it's out of particles is quite easy to create.

Regarding the setting of the gravity in the object parameters we have to think about how we would handle the transition from the current way (X and Y) to the new way (intensity and angle). Maybe we can add a checkbox to trigger one way or the other
